salt/client/ssh/__init__.py

```diff
@@ -1075,8 +1075,22 @@ def run_wfunc(self):
                     opts_pkg['grains'][grain] = self.target['grains'][grain]
 
             popts = {}
-            popts.update(opts_pkg['__master_opts__'])
             popts.update(opts_pkg)
+
+            # Master centric operations such as mine.get must have master option loaded.
+            # The pillar must then be compiled by passing master opts found in opts_pkg['__master_opts__']
+            # which causes the pillar renderer to loose track of salt master options
+            #
+            # Depending on popts merge order, it will overwrite some options found in opts_pkg['__master_opts__']
+            master_centric_funcs = [
+              "pillar.items",
+              "mine.get"
+            ]
+
+            # Pillar compilation is a master centric operation.
+            # Master options take precedence during Pillar compilation
+            popts.update(opts_pkg['__master_opts__'])
```

Comment:
Q: are the **keys** of the `opts_pkg` aren't the same as in `__master_opts__`? I.e. aren't `opts_pkg` just do not override the master opts? In this case the first `update(opts_pkg)` is not really needed. But I am not entirely sure here.